components/lib/calendar/Calendar.js

```diff
@@ -2489,8 +2489,18 @@ export const Calendar = React.memo(
         }, [props.onViewDateChange, props.value]);
 
         useUpdateEffect(() => {
-            if (previousValue !== props.value && (!viewStateChanged.current || visible)) {
-                updateInputfield(props.value);
+            const newDate = props.value;
+
+            if (previousValue !== newDate) {
+                updateInputfield(newDate);
+
+                // #3516 view date not updated when value set programatically
+                if (!visible && newDate) {
+                    validateDate(newDate);
+                    setViewDateState(newDate);
+                    setCurrentMonth(newDate.getMonth());
```

Comment:
@melloware When the `selectionMode` property of the `Calendar` component is `range` the value of the `newDate` variable is an array and the function `getMonth()` is not available.

Reply:
is already fixed check master